### PR TITLE
[SPARK-50036][CORE][PYTHON] Include SPARK_LOG_SCHEMA in the context of REPL shell

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Before the Change:

Users needed to import LOG_SCHEMA to read structured logs as a JSON data source:
```
import org.apache.spark.util.LogUtils.LOG_SCHEMA

val logDf = spark.read.schema(LOG_SCHEMA).json("path/to/logs")
 ```

After the Change:

- Renamed for Clarity:`LOG_SCHEMA` has been renamed to `SPARK_LOG_SCHEMA` to make its purpose more clear.
- No Import Needed in REPL Shells: You can now use `SPARK_LOG_SCHEMA` directly in REPL environments like spark-shell and pyspark without importing it.

Now, you can read structured logs without the import:

```
val logDf = spark.read.schema(SPARK_LOG_SCHEMA).json("path/to/logs")
```

### Why are the changes needed?

Simply the way to query structured logs

### Does this PR introduce _any_ user-facing change?

No, this a new feature in Spark 4.0
### How was this patch tested?

Existing tests
### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: common/utils/src/main/scala/org/apache/spark/util/LogUtils.scala [hunk 1]
```diff
@@ -29,9 +29,9 @@ object LogUtils {
   /**
    * Schema for structured Spark logs.
    * Example usage:
-   *   val logDf = spark.read.schema(LOG_SCHEMA).json("path/to/logs")
+   *   val logDf = spark.read.schema(SPARK_LOG_SCHEMA).json("path/to/logs")
    */
-  val LOG_SCHEMA: String = """
+  val SPARK_LOG_SCHEMA: String = """
```
Comment (dongjoon-hyun) [inline common/utils/src/main/scala/org/apache/spark/util/LogUtils.scala:34]: Well, may I ask what kind of confusions you have in your mind? This is already `org.apache.spark.util.LogUtils.LOG_SCHEMA`, isn't it?

**Review by dongjoon-hyun — APPROVED** [review]
For the record, I'm not against about this change. 

Thank you for this tidy-up before Apache Spark 4.0.0 release (February 2025), @gengliangwang .

File: common/utils/src/main/scala/org/apache/spark/util/LogUtils.scala (hunk 1 above)
Comment (gengliangwang) [reply common/utils/src/main/scala/org/apache/spark/util/LogUtils.scala:34]: After the change, there is no need to import `org.apache.spark.util.LogUtils.LOG_SCHEMA` in repl shell. If we are going to expose this by default in `bin/spark-shell` and `bin/pyspark`, we had better to make it more specific.

File: python/pyspark/shell.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
 
 import pyspark
 from pyspark.core.context import SparkContext
+from pyspark.logger import SPARK_LOG_SCHEMA
```
Comment (zeal-thinker) [inline python/pyspark/shell.py:33]: Where is SPARK_LOG_SCHEMA used in this file? What am I missing?

File: python/pyspark/logger/logger.py [hunk 3]
```diff
@@ -20,6 +20,16 @@
 import json
 from typing import cast, Optional
 
+SPARK_LOG_SCHEMA = (
```
Comment (zeal-thinker) [inline python/pyspark/logger/logger.py:23]: Why did you move SPARK_LOG_SCHEMA from util.LogUtils to logger? SPARK_LOG_SCHEMA is in util.LogUtils for scala.

File: python/pyspark/shell.py (hunk 2 above)
Comment (gengliangwang) [reply python/pyspark/shell.py:33]: it is import in the pyspark shell so that users can use the constant directly.

File: python/pyspark/logger/logger.py (hunk 3 above)
Comment (gengliangwang) [reply python/pyspark/logger/logger.py:23]: There is no backward compatibility issue here. This is new in Spark 4. I think putting in the logger is more here is more appropriate。

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by LuciferYang** [comment]
@gengliangwang Python linter failed:

```
Run PYTHON_EXECUTABLE=python3.9 ./dev/lint-python
starting python compilation test...
python compilation succeeded.

starting black test...
black checks failed:
would reformat /__w/spark/spark/python/pyspark/shell.py

Oh no! 💥 💔 💥
1 file would be reformatted, 1066 files would be left unchanged.
Please run 'dev/reformat-python' script.
1
```

**Comment by gengliangwang** [comment]
@dongjoon-hyun @HyukjinKwon @LuciferYang @zeal-thinker thanks for the review. Merging to master
